client/dashboard/app/router.tsx

```diff
@@ -155,7 +156,10 @@ const siteSettingsSiteVisibilityRoute = createRoute( {
 	path: 'settings/site-visibility',
 	loader: async ( { params: { siteSlug } } ) => {
 		const site = await queryClient.ensureQueryData( siteBySlugQuery( siteSlug ) );
-		queryClient.ensureQueryData( siteSettingsQuery( site.ID ) );
+		Promise.all( [
+			queryClient.ensureQueryData( siteSettingsQuery( site.ID ) ),
+			queryClient.ensureQueryData( siteDomainsQuery( site.ID ) ),
```

Comment:
We only need to ensure this query when `site.launch_status` is not `'unlaunched'`

https://github.com/Automattic/wp-calypso/blob/2830481c60e241a716ab248b871b7196b9e383aa/client/dashboard/sites/settings-site-visibility/index.tsx#L52

I wonder if we should also gate it here, or if that's too much business logic in the loader 🤔